src/plone/importexport/utils.py

```diff
@@ -1,22 +1,47 @@
 # -*- coding: UTF-8 -*-
-from bs4 import BeautifulSoup
-from plone.importexport.exceptions import ImportExportError
-from plone.uuid.interfaces import IUUID
-
-import cStringIO
 import csv
+import cStringIO
 import fnmatch
 import json
 import logging
 import operator
 import os
-import StringIO
 import zipfile
+import StringIO
 
+from bs4 import BeautifulSoup
+from plone import api
+from plone.uuid.interfaces import IUUID
+
+from plone.importexport.exceptions import ImportExportError
+from plone.importexport.interfaces import IImportExportSettings
 
 log = logging.getLogger()
 
 
+def remove_from_list(ls, val):
+    if val in ls:
+        ls.remove(val)
+
+
+def get_metadata_pKeys():
+    """ Get metadata fields that are allowed to be used as the primary key.
+    """
+    
+    catalog = api.portal.get_tool('portal_catalog')
+    default_ = set(catalog.indexes())
```

Comment:
@ironmaniiith, can you change this to:
```
default_ = set(catalog._catalog.names)
```
Plone catalog indexes do not equal to metadata. I think at the time, I didn't know how to retrieve all the metadata, therefore, I had use indexes instead. Please test if the set(catalog._catalog.names) works before you push.